docs/adrs/003-Retry_endpoint_design.md

```diff
@@ -0,0 +1,162 @@
+
+#### Data inconsistency
+* A new `retry_ids` field will show a list of jobs that have been retried using this parent id. Retry_count will be returned as a calculated field based off of retry_ids
+* `retry_toggle` field will allow a seperate process to check for jobs that didn't finish the entire retry lifecycle:
+1) Launch child jobs
+2) Notify the batch parent of the child,
+3) Notify the retry parent of the child,
+4) Update the retry_toggle field
+
+#### Won't do
+* Prevent multiple in-flight retries of the same original job to prevent the user from wasting their own resources (and the queues resources)
+* Add retry_number field
+
+## New priority
+* Create a retry_jobs field, and expose list in api, and a T/F completeness toggle
+* Non blocking job submission / (Possibly htcondor submit)
+* Add failure conditions in run method
+* Add thread to perform actions based on toggle
+
+
+### Questions
+
+#### Answered:
+#### Q: should the number of retries of a job be limited, and if so, where? e.g. a max_retries field in the parent job? wait and see whether people attempt to rerun jobs that have already failed nine zillion times?
+A: Make a ticket for this and add to backlog
+
+#### Q: How do we prevent jobs with identical parameters from being rerun more than once within a retry_jobs request?
+A: We have decided to allow multiple jobs with the same params to be re-run in the same `retry_jobs` request.
+
+#### Q: How do we find the most recent retry of a job?
+A: The client using the ee2 API would have to figure it out using the `retry_parent` and job creation date fields. (Unless we added other fields to help with this)
+
+#### Q: How do we ensure that the app version is correctly run each time when submitting from the narrative?
+A: We would need to change the narrative to submit the git commit hash instead of a version tag
+
+#### Q: How do we handle DB consistency during retry failure? 
+Looks like the options are
+* implement db integrity checks and two-phase commits for making the relationships between a job, its `retry_parent`, and the batch container
+* accept that the db info may be incomplete and write workarounds into the clients
+* (upgrade to Mongo 4.4 for better transaction support)
+
+
+##### Q: Do we want to support ResourceRequirements
+A: Probably not in the short term
+
+
+#### Q: how to prevent incorrect parent-child relationships being created -- should the client be allowed to specify a parent ID? Is it currently possible to add a new child to a parent job if the child is a new job, rather than an existing job ID / set of params that is being rerun?
+A: Not necessarily relevant to this endpoint, more of a `run_job_batch` endpoint question. Currently the `retry_parent` and `parent_job_id` are looked up from the ee2 record on retry, and not specified in this endpoint.
+
+#### Shorter Q and A
+
+    Should we track a retry count? (Done)
+    Should users see this retry count? (Unknown TBD)
```

Comment:
It's visible right now so...

Reply:
It's not visible to the user in the UI

Reply:
This is about EE2 though, so I presume we're talking about the EE2 API.

Reply:
This was a discussion question that we discussed in the context of the UI and also in the context of the api endpoint. So what do you want to be done here?

Reply:
How about 
```
A: Visible in the EE2 API, UI is TBD
```

or something like that

Reply:
ok done.

Reply:
:+1: